### PR TITLE
Remove json writer

### PR DESCRIPTION
**Pull request by matthiasbeyer**
## Proposed changes

So while browsing the code I found that the `json_writer` crate is entirely unnecessary.
What I did here was replace it with normal `serde_json` code.

Because of this I was able to alter tests:

* I removed `serialize_unexpected_end_of_message` because the state it tested (object not closed) is not reachable anymore
* I changed `serialize_multi_value_message` to not test for string-equality, but whether values are actually there as expected.

I'm not 100% satisfied with the name `InnerJson` for the serialization helper type, but as this is not `pub`, I don't worry too much about it.

I do believe that the changes from 10ec1aadb91fc9cd1503b8b07c9f48d503193aeb and 4094a2582aff3b8859dd6b44d77b521064b8966a can further be enhanced by moving the type definitions in a common crate that contains the `InnerJson` type I used.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

---

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Lets see what CI can tell us...

**Review by didier-wenzek — COMMENTED** [review]
The goal of this `json_writer` was to avoid an intermediate memory representation similar to `InnerJson` (along the chain of transformations from a string representation of some measurements and the target string representation of the target c8y measurement). Somehow the json-writer is the complement of the measurement visitor. The aim of this pair were to directly parse an input string into a translated output string. A goal only partially reached, notably because of the need to group related measurements.

I agree that this optimization was premature and that the memory footprint can surely be improved in many other places.

One point is sure. The json-writer optimization makes less sense with the new planned architecture where the rust components exchange rust structured messages. At least one intermediate in-memory representation is then required and there is no more point in trying to directly convert MQTT in-messages into out-messages.

In summary this PR makes sense ... but I need some time to digest it.

**Comment by didier-wenzek** [comment]
Closing as this raises issues while not being a priority
